lib/validator/validator.go

```diff
@@ -406,25 +407,49 @@ func validateRule(v *validator.Validate, structLevel *validator.StructLevel) {
 		}
 	}
 
-	// Check for mismatch in IP versions
-	if rule.Source.Net != nil && rule.IPVersion != nil {
-		if rule.Source.Net.Version() != *(rule.IPVersion) {
-			structLevel.ReportError(reflect.ValueOf(rule.Source.Net), "Source.Net",
-				"", reason("rule contains an IP version that does not match src CIDR version"))
-		}
+	// Check that only one of the net or nets fields is specified.
```

Comment:
Given our desire to ween people off the singular, I wonder if we should be a little more strict and reject if there is a mix of any of the Net and Nets fields (e.g.  Source.Net  && Dest.NotNets)

WDYT?